src/main/java/edu/hm/hafner/analysis/IssueFilterBuilder.java

```diff
@@ -0,0 +1,281 @@
+package edu.hm.hafner.analysis;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import edu.hm.hafner.analysis.Issue;
+
+/**
+ * Builds a IssueFilter 
+ * @author Raphael Furch
+ */
+public class IssueFilterBuilder {
+
+    /**
+     * List of include filters.
+     */
+    private Collection<Predicate<Issue>> filterInclude = new ArrayList<>();
+
+    /**
+     * List of exclude filters.
+     */
+    private Collection<Predicate<Issue>> filterExclude = new ArrayList<>();
+
+    /**
+     * Include combine behavior.
+     * true = or.
+     * false = and.
+     */
+    private boolean includeCombineWithOr = true;
+    /**
+     * Exclude combine behavior.
+     * true = or.
+     * false = and.
+     */
+    private boolean excludeCombineWithOr = false;
```

Comment:
Ich würde mich für eine Möglichkeit entscheiden und den Switch rauswerfen. So wird auch unten an einigen Stellen vereinfacht. -> YAGNI

Reply:
Danke für das Feedback. 
Hier war ja die Diskussion, was logischer / sinnvoller ist. Gab es dazu schon eine finale Entscheidung von Herrn Hafner?

Reply:
Ja, die Entscheidung gibt es in Moodle: ODER Vernüpfung. Ich denke auch, dass die Auswahl entfallen kann und damit der Code vereinfacht werden kann.